components/script_bindings/weakref.rs

```diff
@@ -80,6 +81,20 @@ pub trait WeakReferenceable: DomObject + Sized {
     }
 }
 
+impl<T: WeakReferenceable> Eq for WeakRef<T> {}
+
+impl<T: WeakReferenceable> Hash for WeakRef<T> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        unsafe {
```

Comment:
Keep it safe by [using the domroot](https://github.com/servo/servo/blob/0b92caacaf802d0084111560c8548c21293ab87e/components/script_bindings/root.rs#L355)?

Needs additional review cc @jdm